Kitodo-FileManagement/src/main/java/org/kitodo/filemanagement/locking/UpgradeableReadLock.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.filemanagement.locking;
+
+import java.net.ProtocolException;
+import java.net.URI;
+import java.nio.file.AccessDeniedException;
+
+import org.kitodo.api.filemanagement.LockingMode;
+
+/**
+ * A read lock that may, under certain circumstances, receive a one-time upgrade
+ * to write.
+ */
+class UpgradeableReadLock extends AbstractLock {
+    /**
+     * This lock is so complicated that it has been implemented as a state
+     * machine. This variable contains the current state.
+     */
+    private UpgradeableReadLockState state;
+
+    /**
+     * Creates a new upgradeable read lock.
+     *
+     * @param owner
+     *            future owner of the lock
+     * @param uri
+     *            locked URI
+     * @param streamManagement
+     *            the stream management
+     * @param immutableReadFileManagement
+     *            the immutable read file management
+     * @param write
+     *            whether the lock is currently in write mode
+     */
+    UpgradeableReadLock(String owner, URI uri, StreamManagement streamManagement,
```

Comment:
StreamManagement streamManagement and ImmutableReadFileManagement immutableReadFileManagement are not used.